File: frontend/apps/web-antd/src/api/site.ts
/**
 * 站点基础信息 API（系统管理-基础信息配置页 + 登录页公开渲染）
 *
 * 对接后端 backend/app/api/v1/endpoints/site.py：
 * - GET /site/basic-info  公开免登录（登录页渲染公司简称/封面 LOGO）
 * - GET /configs/site     登录可读完整字段（配置页加载）
 * - PUT /configs/site     ADMIN 更新（配置页保存）
 * - POST /site/logo       ADMIN 上传 LOGO 图片（multipart）
 *                          query type=cover|content 区分封面/内容页 LOGO
 */
import { requestClient } from '#/api/request';

export namespace SiteApi {
  /** LOGO 类型：cover=封面页横向 / content=内容页方形 */
  export type LogoType = 'content' | 'cover';

  /** 公开基础信息（登录页所需：公司简称 + 封面 LOGO） */
  export interface BasicInfoPublic {
    companyFullName: string;
    companyShortName: string;
    /** 封面页 LOGO（横向，登录页左上角） */
    coverLogoUrl: string;
  }

  /** 完整基础信息（配置页加载） */
  export interface SiteConfig {
    companyFullName: string;
    companyShortName: string;
    /** 内容页 LOGO（方形，主布局左上角） */
    logoUrl: string;
    /** 封面页 LOGO（横向，登录页左上角） */
    coverLogoUrl: string;
    contactPerson: string;
    contactPhone: string;
    contactEmail: string;
    address: string;
    authorizedLoopCount: null | number;
    licenseExpireDate: null | string;
    systemDeployId: string;
    systemDeployDate: null | string;
    serviceProvider: string;
  }

  /** 基础信息更新体（全可选，仅传需修改的字段） */
  export type SiteConfigUpdate = Partial<SiteConfig>;

  /** LOGO 上传响应 */
  export interface LogoUploadResponse {
    url: string;
  }
}

/** 公开基础信息（免登录，登录页调用） */
export function getSiteBasicInfoApi() {
  return requestClient.get<SiteApi.BasicInfoPublic>('/site/basic-info');
}

/** 完整基础信息（登录可读，配置页加载） */
export function getSiteConfigApi() {
  return requestClient.get<SiteApi.SiteConfig>('/configs/site');
}

/** 更新基础信息（ADMIN） */
export function updateSiteConfigApi(data: SiteApi.SiteConfigUpdate) {
  return requestClient.put<SiteApi.SiteConfig>('/configs/site', data);
}

/** 上传 LOGO 图片（ADMIN，multipart）
 * @param file 图片文件
 * @param type LOGO 类型：cover=封面页（横向）/ content=内容页（方形）
 */
export function uploadLogoApi(file: File, type: SiteApi.LogoType = 'content') {
  const formData = new FormData();
  formData.append('file', file);
  return requestClient.post<SiteApi.LogoUploadResponse>(
    `/site/logo?type=${type}`,
    formData,
    {
      headers: { 'Content-Type': 'multipart/form-data' },
    },
  );
}

File: backend/app/schemas/site.py
"""站点基础信息配置 Schema（IDS v3.2 §2.10 扩展）.

承载客户/部署方的基础信息：公司名/LOGO/联系人/授权回路数等，
登录页通过公开接口读取公司简称/全称/LOGO 渲染，ADMIN 可在
系统管理-基础信息页编辑全部字段。

存储：sys_config 单条 JSON key ``site.basic_info``（见 endpoints/site.py）。

LOGO 字段说明：
- ``logoUrl``：内容页 LOGO，**方形**（建议 128×128 ~ 256×256），
  显示在主布局每个页面左上角侧边栏顶部（VbenLogo 32×32 渲染）。
- ``coverLogoUrl``：封面页 LOGO，**横向布局**（建议 2:1~4:1 宽高比，
  如 240×80），显示在登录页左上角。
"""

from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field


class SiteBasicInfoPublic(BaseModel):
    """公开基础信息（免登录，登录页渲染所需）."""

    companyFullName: str = Field(default="", description="公司全称")
    companyShortName: str = Field(default="", description="公司简称")
    # 登录页左上角展示的横向 LOGO（封面页专用）
    coverLogoUrl: str = Field(default="", description="封面页 LOGO URL（横向布局，登录页用）")


class SiteConfigResponse(BaseModel):
    """完整基础信息（登录后 ADMIN 可读）."""

    companyFullName: str = Field(default="", description="公司全称")
    companyShortName: str = Field(default="", description="公司简称")
    # 内容页 LOGO（方形，主布局左上角）
    logoUrl: str = Field(default="", description="内容页 LOGO URL（方形，主布局左上角）")
    # 封面页 LOGO（横向，登录页左上角）
    coverLogoUrl: str = Field(default="", description="封面页 LOGO URL（横向布局，登录页用）")
    contactPerson: str = Field(default="", description="联系人")
    contactPhone: str = Field(default="", description="联系电话")
    contactEmail: str = Field(default="", description="联系邮箱")
    address: str = Field(default="", description="公司地址")
    authorizedLoopCount: int | None = Field(default=None, description="授权回路数量")
    licenseExpireDate: date | None = Field(default=None, description="授权到期日期")
    systemDeployId: str = Field(default="", description="系统部署编号")
    systemDeployDate: date | None = Field(default=None, description="系统部署日期")
    serviceProvider: str = Field(default="", description="服务提供方")


class SiteConfigUpdateRequest(BaseModel):
    """基础信息更新请求（全字段可选，ADMIN 提交）."""

    companyFullName: str | None = Field(default=None, description="公司全称")
    companyShortName: str | None = Field(default=None, description="公司简称")
    logoUrl: str | None = Field(default=None, description="内容页 LOGO URL（方形）")
    coverLogoUrl: str | None = Field(default=None, description="封面页 LOGO URL（横向）")
    contactPerson: str | None = Field(default=None, description="联系人")
    contactPhone: str | None = Field(default=None, description="联系电话")
    contactEmail: str | None = Field(default=None, description="联系邮箱")
    address: str | None = Field(default=None, description="公司地址")
    authorizedLoopCount: int | None = Field(default=None, description="授权回路数量")
    licenseExpireDate: date | None = Field(default=None, description="授权到期日期")
    systemDeployId: str | None = Field(default=None, description="系统部署编号")
    systemDeployDate: date | None = Field(default=None, description="系统部署日期")
    serviceProvider: str | None = Field(default=None, description="服务提供方")


class LogoUploadResponse(BaseModel):
    """LOGO 上传响应."""

    url: str = Field(description="LOGO 访问 URL（相对路径，前端拼 API base）")


__all__ = [
    "LogoUploadResponse",
    "SiteBasicInfoPublic",
    "SiteConfigResponse",
    "SiteConfigUpdateRequest",
]
